_source/_posts/2022-03-29-github-actions-graalvm.adoc

```diff
@@ -0,0 +1,490 @@
+
+*Prerequisites*:
+
+- https://sdkman.io[Java 11]+
+- https://nodejs.org[Node 16]
+- A https://github.com/signup[GitHub Account]
+
+toc::[]
+
+You can read the full conversation (aka a condensed version of this post) via the tweet below, or keep reading to walk through the trials and tribulations I experienced with GitHub Actions and GraalVM. Hopefully my learnings will save you hours of time.
+
+++++
+{% twitter 1496278460293324801 %}
+++++
+
+== Configure a Full-Stack Native Java app to use GitHub Actions
+// == Configure a JHipster app to use GitHub Actions
+// == Configure a Spring Native app to use GitHub Actions
+
+I'm going to speed things up for you and just show you how to configure GitHub Actions for an existing JHipster app. In this case, it's a full-stack React + Spring Boot app.
+
+====
+If you'd like a bit more background, please read https://auth0.com/blog/full-stack-java-with-react-spring-boot-and-jhipster/[Full Stack Java with React, Spring Boot, and JHipster] followed by link:/blog/2022/03/03/spring-native-jhipster[Introducing Spring Native for JHipster].
+====
+
+Clone the result of these two blog posts, right after I integrated the JHipster Native blueprint 1.0. Install its dependencies using npm.
+
+[source,shell]
+----
+git clone -b jhipster-native-1.0 \
+  https://github.com/oktadev/auth0-full-stack-java-example.git flickr2
+cd flickr2
+npm install
+----
+
+Then, create a https://github.com/new[new repo on GitHub]. For example, `jhipster-flickr2`.
+
+Next, push this example project to it.
+
+[source,shell]
+----
+USERNAME=<your-github-username>
+git remote rm origin
+git remote add origin git@github.com:${USERNAME}/jhipster-flickr2.git
+git branch -M main
+git push -u origin main
+----
+
+== Automate the wait with JHipster's CI/CD
+
+Building native images with GraalVM brings me back to the days when we'd build Java apps with Ant and XDoclet back in the early 2000s. You'd start the build and go do something else for a while because it took _minutes_ for the artifacts to be built.
+
+There's another often-overlooked issue with native binaries: you have to build one for each operating system. It's not like Java where you can build a JAR and run it anywhere.
+
+Generate continuous integration (CI) configuration for GitHub Actions using https://www.jhipster.tech/setting-up-ci/[JHipster's CI/CD sub-generator].
+
+[source,shell]
+----
+jhipster ci-cd
+----
+
+This will prompt you to select a CI/CD pipeline. Select **GitHub Actions**.
+
+image::{% asset_path 'blog/github-actions-graalvm/jhipster-ci-cd.png' %}[alt=Welcome to the JHipster CI/CD Sub-Generator,width=800, align=center]
+
+When prompted for the tasks/integrations (Sonar, Docker, Snyk, Heroku, and Cypress Dashboard), don't select any for this quick example. This will create a `.github/workflows/github-actions.yml` file. Rename it to `.github/workflows/main.yml` for simplicity.
```

Comment:
```suggestion
When prompted for the tasks/integrations for this quick example (Sonar, Docker, Snyk, Heroku, and Cypress Dashboard), don't select any. This will create a `.github/workflows/github-actions.yml` file. Rename it to `.github/workflows/main.yml` for simplicity.
```
reads better imo, because it puts some distance btwn "this" and "this"